server/src/main/kotlin/io/spine/examples/pingh/server/Application.kt

```diff
@@ -0,0 +1,190 @@
+ * from an external clock or a system scheduler.
+ *
+ * In non-production mode, the following actions are performed during initialization:
+ *
+ * 1. Configures the environment for local use, including loading
+ * GitHub App secrets from the configuration file.
+ *
+ * 2. Initializes an [IntervalClock] to emit an event with the current time
+ * to the server every second.
+ */
+internal class Application {
+
+    internal companion object {
+        /**
+         * The port on which the Pingh server runs.
+         */
+        private const val pinghPort = 50051
+    }
+
+    /**
+     * The Pingh server.
+     */
+    internal val server: Server
+
+    init {
+        configureEnvironment()
+        startClock()
+        server = createServer()
+    }
+
+    /**
+     * Configures the server environment.
+     */
+    private fun configureEnvironment() {
+        ServerEnvironment.`when`(DefaultMode::class.java)
+            .useStorageFactory { InMemoryStorageFactory.newInstance() }
+            .useDelivery { Delivery.localAsync() }
+            .useTransportFactory { InMemoryTransportFactory.newInstance() }
+
+        ServerEnvironment.`when`(Production::class.java)
+            .useStorageFactory { DatastoreStorageFactory.remote() }
+            .useDelivery { Delivery.localAsync() }
+            .useTransportFactory { InMemoryTransportFactory.newInstance() }
+    }
+
+    /**
+     * Starts emitting periodic events with the current time to the server.
+     *
+     * In [Production] mode, starts a [server][startHeartbeatServer]
+     * to handle HTTP requests from an external clock or system scheduler.
+     * In non-production mode, starts a [clock][IntervalClock]
+     * to emit an event to the server every second.
+     */
+    private fun startClock() {
+        if (isProduction()) {
+            startHeartbeatServer(Clock())
+        } else {
+            IntervalClock(1.seconds).start()
+        }
+    }
+
+    /**
+     * Creates a new Spine `Server` instance at the [pinghPort].
+     *
+     * The server includes bounded contexts of [Sessions][io.spine.examples.pingh.sessions]
+     * and [Mentions][io.spine.examples.pingh.mentions].
+     */
+    private fun createServer(): Server {
+        val httpEngine = CIO.create()
+        return Server
+            .atPort(pinghPort)
+            .add(
+                newSessionsContext(
+                    RemoteGitHubAuthentication(gitHubApp(), httpEngine),
+                    RemoteGitHubUsers(httpEngine)
+                )
+            )
+            .add(newMentionsContext(RemoteGitHubSearch(httpEngine)))
+            .build()
+    }
+
+    /**
+     * Returns the GitHub App secrets required to make authentication requests
+     * on behalf of the App.
+     *
+     * In [Production] mode, obtains secrets from the Secret Manager.
+     * In non-production mode, loads secrets from resource folder.
+     */
+    private fun gitHubApp(): GitHubApp =
```

Comment:
Can we please move the creation/loading of `GitHubApp` instance out of the `Application` class, but still keep it in `Application.kt`? These two methods (`gitHubApp()` and `loadGitHubAppSecrets()` are tightly coupled, and aren't really related to the `Application` instance itself.